src/System.Data.SqlClient/tests/ManualTests/SQL/AdapterTest/AdapterTest.cs

```diff
@@ -0,0 +1,1505 @@
+                }
+                ExecuteNonQueryCommand("DROP TABLE varbin");
+                conn.Close();
+            }
+        }
+
+        [CheckConnStrSetupFact]
+        public void UpdateOffsetTest()
+        {
+            string createTable = "CREATE TABLE varbin(cvarbin VARBINARY(7000), cimage IMAGE)";
+            ExecuteNonQueryCommand(createTable);
+
+            bool dropSP = false;
+
+            SqlConnection conn = new SqlConnection(DataTestUtility.TcpConnStr);
+            try
+            {
+                string createSP =
+                    "CREATE PROCEDURE sp_insertvarbin (@val_cvarbin VARBINARY(7000), @val_cimage IMAGE)" +
+                    "AS INSERT INTO varbin (cvarbin, cimage)" +
+                    "VALUES (@val_cvarbin, @val_cimage)";
+                ExecuteNonQueryCommand(createSP);
+                dropSP = true;
+
+                int rowsAffected = 0;
+
+                SqlDataAdapter sa = new SqlDataAdapter();
+
+                SqlCommand cmdInsert = new SqlCommand("sp_insertvarbin", conn);
+                SqlCommand cmdSelect = new SqlCommand("select * from varbin", conn);
+
+                cmdInsert.CommandType = CommandType.StoredProcedure;
+                conn.Open();
+
+                SqlParameter p1 = cmdInsert.Parameters.Add(new SqlParameter("@val_cvarbin", SqlDbType.VarBinary, 7000));
+                SqlParameter p2 = cmdInsert.Parameters.Add(new SqlParameter("@val_cimage", SqlDbType.Image, 7000));
+                DataSet ds = new DataSet();
+
+                SqlCommand tc = new SqlCommand("delete varbin", conn);
+                tc.ExecuteNonQuery();
+
+                byte[] b = new byte[7];
+                b[0] = 0x01;
+                b[1] = 0x02;
+                b[2] = 0x03;
+                b[3] = 0x04;
+                b[4] = 0x05;
+                b[5] = 0x06;
+                b[6] = 0x07;
+                p1.Value = b;
+                p1.Size = 4;
+
+                p2.Value = b;
+                p2.Size = 3;
+                p2.Offset = 4;
+                rowsAffected = cmdInsert.ExecuteNonQuery();
+                sa.SelectCommand = cmdSelect;
+                sa.Fill(ds, "goofy");
+
+                byte[] expectedBytes1 = { 0x01, 0x02, 0x03, 0x04 };
+                byte[] val = (byte[])(ds.Tables[0].Rows[0][0]);
+                Assert.True(ByteArraysEqual(expectedBytes1, val), "FAILED: Test 1: Unequal byte arrays.");
+
+                byte[] expectedBytes2 = { 0x05, 0x06, 0x07 };
+                val = (byte[])(ds.Tables[0].Rows[0][1]);
+                Assert.True(ByteArraysEqual(expectedBytes2, val), "FAILED: Test 2: Unequal byte arrays.");
+            }
+            finally
+            {
+                if (dropSP)
+                {
+                    ExecuteNonQueryCommand("DROP PROCEDURE sp_insertvarbin");
+                }
+                ExecuteNonQueryCommand("DROP TABLE varbin");
+                conn.Close();
```

Comment:
A general note about the cleanup for all the tests: can you close the connections in the tests in the try block and open a new connection for cleanup in finally? The reason I ask for this change is so that a bad connection due to an exception during command execution will try to cleanup and there is a chance that cleanup will fail. Since the connections are in the pool by default there shouldn't be an overhead with close() and open()